### PR TITLE
mutagen: 0.17.1 -> 0.17.6

### PR DESCRIPTION
**Pull request by dpk**
Changelog: see https://github.com/mutagen-io/mutagen/releases

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] ~~(Package updates) Added a release notes entry if the change is major or breaking~~
  - [ ] ~~(Module updates) Added a release notes entry if the change is significant~~
  - [ ] ~~(Module addition) Added a release notes entry if adding a new NixOS module~~
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 349290` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mutagen</li>
  </ul>
</details>

File: pkgs/tools/misc/mutagen/default.nix [hunk 1]
```diff
@@ -2,16 +2,16 @@
 
 buildGoModule rec {
   pname = "mutagen";
-  version = "0.17.1";
+  version = "0.17.6";
```
Comment (phanirithvij) [inline pkgs/tools/misc/mutagen/default.nix:5]:
```suggestion
  version = "0.18.0";
```

File: pkgs/tools/misc/mutagen/default.nix [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
 
   src = fetchFromGitHub {
     owner = "mutagen-io";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-M7h8qlqqGK4Nl4yXL7ZhGTq/CL+LdDpI/nv90koyu3Y=";
+    hash = "sha256-GkTMc+gS90M0Jx00nV8+QW/F2iBMmi/oNpwFun18yy4=";
```
Comment (phanirithvij) [inline pkgs/tools/misc/mutagen/default.nix:11]:
```suggestion
    hash = "sha256-/UigWQMk+VDMGna/ixctU8MR7VNPpOTOGNUtuYx8DS0=";
```

File: pkgs/tools/misc/mutagen/default.nix [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
   };
 
-  vendorHash = "sha256-kfzT+230KY2TJVc0qKMi4TysmltZSgF/OvL5nPLPcbM=";
+  vendorHash = "sha256-Bd2fI2p/8JNoN3Q3sJFuF83YyXUi5j68udW6QnVMViM=";
```
Comment (phanirithvij) [inline pkgs/tools/misc/mutagen/default.nix:14]:
```suggestion
  vendorHash = "sha256-J92LzjIsLlBOhnkWrp8MRgoe+4NzXyBgqQRigse5GQk=";
```

File: pkgs/tools/misc/mutagen/default.nix [hunk 4]
```diff
@@ -21,7 +21,7 @@ buildGoModule rec {
     postFetch = ''
       rm $out/mutagen # Keep only mutagen-agents.tar.gz.
     '';
-    hash = "sha256-RFB1/gzLjs9w8mebEd4M9Ldv3BrLIj2RsN/QAIJi45E=";
+    hash = "sha256-0Oh5k2tKhQzlL8E6vWEMDyZ1JGw329K5E7Y8kam2Sec=";
```
Comment (phanirithvij) [inline pkgs/tools/misc/mutagen/default.nix:24]:
```suggestion
    hash = "sha256-EGMBsv6WjmWj/tOhtOORd6eqHmdfJb5pxPrb3zr/ynI=";
```

**Review by phanirithvij — COMMENTED** [review]
0.18.0

**Pull request closed without merging**